proposals/20210524-versioning-and-release-of-the-libs-artifacts.md

```diff
@@ -0,0 +1,97 @@
+
+New (backward compatible) functionalities would make a minor version bump.
+
+New (backward compatible) fixes would make a patch version bump.
+
+### Steps
+
+1. Come up with all the `falcosecurity/libs` maintainers and decide whether the current APIs should be considered stable or not
+
+Concretely this means the `falcosecurity/libs` maintainers have to decide whether to start versioning from `1.y.z` or from `0.y.z`.
+
+2. Setup [milestones](https://github.com/falcosecurity/libs/milestones) on `falcosecurity/libs`
+
+Same way as in Falco, changes, fixes, and new features have to be grouped in GitHub milestones.
+
+The milestone string has to reflect the upcoming release version.
+
+By looking at the set of merged-in pull requests belonging to a given milestone (ie., release) it can happen that the release version string must change to reflect the above-mentioned SemVer 2.0 rules.
+
+3. Ensure the `falcosecurity/libs` pull request template contains a correct release-note block
+
+4. Ensure the Falco Infra plugin list for `falcosecurity/libs` contains the following plugins: `milestone`, `mergecommitblocker`, `release-note`
+
+5. Ensure the Falco Infra configuration for the `tide` components blocks pull request missing the release notes block (via `do-not-merge/release-note-label-needed` label)
+
+6. Decide with other `falcosecurity/libs` maintainers the tool to generate the CHANGELOG.md
+
+This document proposes to stick with `rn2md` given the release process is very similar to the Falco one and it's based on conventional release notes.
+
+7. Change the CMake files in `falcosecurity/libs` so to compute a version string either from an external variable or from the current git index
+
+8. Add the packaging (DEB, RPM) functionality to the CMake files in `falcosecurity/libs`
+
+9. Create a Falco Infra job to test the libsinp and libscap packages creation
+
+10. Create a Falco Infra job that runs only on git tags on the main branch of `falcosecurity/libs` to create the packages and to ships them into our open repositories at <https://download.falco.org/?prefix=packages>
+
+11. Create another Falco Infra job (running over `falcosecurity/test-infra`) that gets triggered once `falcosecurity/libs` gets released
+
+Its goals are:
+
+- to create the YAML configuration files of the Drivers Build Grid for the current `falcosecurity/libs` version
+- to open a pull request containing those file towards `falcosecurity/test-infra`
+
+Once merged in, this pull request will automatically build the new prebuilt Falco driver artifacts for the current `falcosecurity/libs` version (eg., `0.y.z`) and deploy them to <https://download.falco.org/?prefix=driver/0.y.z>
+
+12. Edit the Falco build system to make it able to link against a system libsinp
+
+13. Write down the release process steps in a RELEASE.md file
```

Comment:
```suggestion
12. Write down the release process steps in a RELEASE.md file
```

I would not steps regarding the Falco build system here since deciding whatever the consumer should do with the artifacts is outside the `libs` scope in my opinion.

Reply:
Agreed, I'll remove point 12 from this proposal